.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,85 @@
+name: CI/CD Pipeline
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - 'App/**'
+      - 'CourseAdmin/**'
```

Comment:
Could you remove CourseAdmin here, since we don't need tests on documentation changes?